lib/azuremetadata/azuremetadatautils.py

```diff
@@ -0,0 +1,95 @@
+from azuremetadata import azuremetadata
+
+
+class QueryException(Exception):
+    pass
+
+
+class AzureMetadataUtils:
+    def __init__(self, data):
+        self._data = data
+        self._parents = {}
+        self._available_params = {}
+        self._parse_data(self._data)
+
+    @property
+    def available_params(self):
+        return self._available_params
+
+    def _parse_data(self, data, parent_key=''):
+        if isinstance(data, list):
+            for item in data:
+                if isinstance(item, dict):
+                    self._parse_data(item, parent_key)
+                else:
+                    raise Exception("List of lists is not supported")
```

Comment:
Do you intentionally raise from the toplevel ? the raise will look like as if it does not belong to the AzureMetaDataUtils namespace unless one reads the stacktrace history

Reply:
Somewhat intentional, so far the metadata API doesn't have a list of lists anywhere in the response -- that exception probably should be an `assert` instead.

Reply:
OK, it is an `assert` now.